### DISCUSSION TITLE
how to configure cpp environment

### DISCUSSION THREAD

**Discussion by destined-wyy** [coder/code-server#7076] [Q&A]
![Screenshot_20241111_183414_com_nightmare_code_VSCodePage](https://github.com/user-attachments/assets/2e597dd5-195f-42f7-aaab-73e7e441465e)
![Screenshot_20241111_183400_com_nightmare_code_VSCodePage](https://github.com/user-attachments/assets/16696359-2b6a-4c91-b7eb-f8098aee8742)

**Comment by code-asher** [comment]
Have you installed the compiler and is it available on `PATH`?

**Reply by code-asher** [reply]
Although, I see in the terminal it did find it.  Why do you have to wrap it in `/bin/bash -c`?  My guess is that it is not set in code-server's environment.  Does it work if you run `gcc` directly?  What does `printenv PATH` and `/bin/bash -c "which gcc"` give you?

**Comment by destined-wyy** [comment]
Thank you for your reply.I'm a beginner in CS field,and have little knowledge about linux os system,so in order to skip the complex configuration,i find a project in github,
https://github.com/nightmare-space/vscode_for_android?tab=readme-ov-file#code-fa



---- Replied Message ----
| From | ***@***.***> |
| Date | 11/13/2024 03:29 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [coder/code-server] how to configure cpp environment (Discussion #7076) |

Although, I see in the terminal it did find it. Why do you have to wrap it in /bin/bash -c? My guess is that it is not set in code-server's environment. Does it work if you run gcc directly? What does printenv PATH and /bin/bash -c "which gcc" give you?

**Reply by code-asher** [reply]
I am not able to test that project, but let me know if you can run these commands in the terminal:

- `which gcc`
- `printenv PATH` 
- `/bin/bash -c "which gcc"`

**Comment by destined-wyy** [comment]
This project wrap code-server,i propose a issue under this project but the author let me propose issue under the code-server project.I try to download the complier in the app code-FA,but failed.



---- Replied Message ----
| From | ***@***.***> |
| Date | 11/13/2024 03:29 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [coder/code-server] how to configure cpp environment (Discussion #7076) |

Although, I see in the terminal it did find it. Why do you have to wrap it in /bin/bash -c? My guess is that it is not set in code-server's environment. Does it work if you run gcc directly? What does printenv PATH and /bin/bash -c "which gcc" give you?

**Comment by destined-wyy** [comment]
Sorry for not checking my messages recently and replying a bit late.



---- Replied Message ----
| From | ***@***.***> |
| Date | 11/15/2024 03:35 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [coder/code-server] how to configure cpp environment (Discussion #7076) |

I am not able to test that project, but let me know if you can run these commands in the terminal:

which gcc
printenv PATH
/bin/bash -c "which gcc"
